### PR TITLE
[SQL-DS-CACHE-165][POAE7-1214]Make compression codec configurable for Spark APE disaggregate mode

### PR DESCRIPTION
**Pull request by yma11**
## What changes were proposed in this pull request?
Make compression codec configurable for Spark APE disaggregate mode

## How was this patch tested?

Q9

### CODE REVIEW

**Comment by yma11** [comment]
@jikunshang @iyupeng please take a review.

File: oap-ape/ape-java/ape-common/src/main/java/com/intel/ape/service/netty/NettyMessage.java [hunk 1]
```diff
@@ -648,7 +651,7 @@ private void writeDataToChannel(ChannelOutboundInvoker out, ChannelPromise promi
                                         ByteBufAllocator allocator, int headerLength,
                                         int originLength) {
             writeHeaderBufToChannel(out, promise, allocator, headerLength,
-                    originLength, originLength, false);
+                    originLength, originLength, false, null);
```
Comment (jikunshang) [inline oap-ape/ape-java/ape-common/src/main/java/com/intel/ape/service/netty/NettyMessage.java:654]: safe to use null? use empty string instead?

**Review by jikunshang — COMMENTED** [review]
Overall LGTM.

File: oap-ape/ape-java/ape-common/src/main/java/com/intel/ape/service/netty/NettyMessage.java (hunk 1 above)
Comment (yma11) [reply oap-ape/ape-java/ape-common/src/main/java/com/intel/ape/service/netty/NettyMessage.java:654]: this value will be only used after compressEnabled checked. so it's okay to give null here.

**Pull request merged**